src/Data/Recruit.ChangeScripts/010_vacancy_backfill_closureReason.js

```diff
@@ -0,0 +1,48 @@
+{
+    const sleepTime = 500;
+    const autoClosure = "Auto";
+    const manualClosure = "Manual";
+
+    function setClosureReason(vac, closureReason) {
+        print(`updating ${vac.vacancyReference}`);
+        db.vacancies.updateOne({ "_id": vac._id }, { $set: { "closureReason": closureReason} });
+    }
+
+    var closedVacancies = db.vacancies.aggregate([
+        { $match : {"status" : "Closed"} },
```

Comment:
As a suggestion, why don't we do two separate queries, one for system closed vacancies, then we batch through them setting the closureReason to `Auto` and another query to batch through the manual.

I assume even the current one is going to take some time and we will need to do a downtime deployment. I think for that reason that half a second sleep is too little.

Reply:
Yes that is another way of doing it, just extra scripting will have to be done. This is simple and it works. I am expecting no more than 2500 closed vacancies by the time we come to release this. Hopefully it should take less than 10 minutes or even quicker. Anyways, don't think we need a down time.